src/components/NavBar/SearchBar.tsx

```diff
@@ -102,8 +104,16 @@ export const SearchBar = () => {
     ...trace,
   }
   const placeholderText = useMemo(() => {
```

Comment:
This will not generate translations, because translations must be statically known. (This is a bug that existed before.)

In order to generate translations, you can't build a string dynamically, but must declare it statically. You also can't dynamically build the `Trans` component. Instead, you'll want your `useMemo` to return a tagged string, like

    t`Search`

Each case will need to return a full string, something like

    const placeholderText = useMemo(() => {
      if (isMobileOrTablet) {
        return t`Search`
      } else {
        if (shouldDisableNFTRoutes) {
          return t`Search tokens`
        } else {
          return t`Search tokens and NFT collections`
        }
      }
    }, [isMobileOrTablet, shouldDisableNFTRoutes])

Lastly, you'll want to put the translated string into the `<Box>` directly, without using a `<Trans>` wrapper.

You can test if this works by extracting translations locally - using `yarn i18n:extract` - and then looking for the extracted strings in the generated `src/locales/en-US.po` file.

Reply:
<img width="394" alt="Screenshot 2023-07-06 at 11 16 11 AM" src="https://github.com/Uniswap/interface/assets/35351983/50a328cd-cc14-489e-ac9d-7fbdbb7731c5">

Is this what I'm looking for in `src/locales/en-US.po`?